pandas/core/generic.py

```diff
@@ -9082,6 +9083,9 @@ def _where(
         name="where",
         name_other="mask",
     )
+    @deprecate_nonkeyword_arguments(
+        version=None, allowed_args=["self", "cond", "other"]
+    )
```

Comment:
In order for the warning to show the correct message, `where` should be defined in `pandas/core/series.py` and `pandas/core/frame.py` as well, and the `deprecate_nonkeyword_arguments` set there as well - see https://github.com/pandas-dev/pandas/pull/41510 for an example of that